pkg/apis/workflow/v1alpha1/workflow_types.go

```diff
@@ -3441,8 +3441,20 @@ type Gauge struct {
 	Value string `json:"value" protobuf:"bytes,1,opt,name=value"`
 	// Realtime emits this metric in real time if applicable
 	Realtime *bool `json:"realtime" protobuf:"varint,2,opt,name=realtime"`
+	// Operation defines the operation to apply with value and the metrics' current value
+	// +optional
+	Operation GaugeOperation `json:"operation,omitempty" protobuf:"bytes,3,opt,name=operation"`
 }
 
+// A GaugeOperation is the set of operations that can be used in a gauge metric.
+type GaugeOperation string
+
+const (
+	GaugeOperationSet GaugeOperation = "Set"
+	GaugeOperationAdd GaugeOperation = "Add"
+	GaugeOperationSub GaugeOperation = "Sub"
+)
```

Comment:
I don't have a strong opinion on this. 
`Inc` and `Dec` can be achieved with Add and Sub. In fact, we could only have `Add`, because `Sub` can also be achieved by setting `Add` with a negative value.